src/registry/middleware/file-uploads.js

```diff
@@ -9,11 +8,17 @@ module.exports = function(req, res, next){
     return next();
   }
 
-  return multer({
-    dest: res.conf.tempDir,
-    fieldSize: 10,
-    rename: function(fieldname, filename){
-      return format('{0}-{1}.tar', filename.replace('.tar', '').replace(/\W+/g, '-').toLowerCase(), Date.now());
```

Comment:
The old filename replacement didn't take in consideration the extension (`.gz`).
The new mechanism has the whole filename so that's why it's slightly different.